third_party/terraform/utils/utils.go.erb

```diff
@@ -79,15 +79,16 @@ func getRouterLockName(region string, router string) string {
 }
 
 func handleNotFoundError(err error, d *schema.ResourceData, resource string) error {
-	if isGoogleApiErrorWithCode(err, 404) {
+		if isGoogleApiErrorWithCode(err, 404) {
 		log.Printf("[WARN] Removing %s because it's gone", resource)
 		// The resource doesn't exist anymore
 		d.SetId("")
 
 		return nil
 	}
 
-	return fmt.Errorf("Error reading %s: %s", resource, err)
+	return errwrap.Wrapf(
+		fmt.Sprintf("Error when reading or editing %s: {{err}}", resource), err)
```

Comment:
This was so confusing when we were debugging, but the handleNotFound error is also called during Deletion.